Sources/Build/BuildDelegate.swift

```diff
@@ -175,12 +176,28 @@ public final class BuildDelegate: BuildSystemDelegate {
     public var outputStream: ThreadSafeOutputByteStream
     public var isVerbose: Bool = false
     public var onCommmandFailure: (() -> Void)?
+    private var numberOfCommandsScanning = 0
```

Comment:
Would wrapping these in a struct help logically group them together, or just add extra complexity?

Thinking something like

```
private struct CommandState {

  // Track progress of various commands based on their [state](https://github.com/apple/swift-llbuild/blob/a989feb2acfe9bcd14d19d1a225c71551e533b0c/products/llbuildSwift/BuildSystemBindings.swift#L300). 
  var scanningCount = 0
  var completedCount = 0
  var startedCount = 0
  var upToDateCount = 0

 var estimatedMaximum: Int {
         return completedCount + scanningCount - upToDateCount
     }
}
```

or something

Reply:
Sounds like a good idea! Will implement it during the next iteration of this PR.

Reply:
I agree, we should wrap all this in a struct.